client/src/components/Event.jsx

```diff
@@ -88,11 +109,23 @@ class Event extends React.Component {
 
     if (this.state.hasPermissionToView) {
       const { id } = this.props.match.params;
-      const { title, description } = this.state;
+      const { title, description, delivery_time } = this.state;
+
+      let LaunchTimeLen = moment(delivery_time).length;

```

Comment:
probably all these calculations for creating `happen` and `timeToLaunch` and `launchDisplay` could be put into method(s) but oh well 💃